File: Dataset/Leetcode/test/14/93.cpp
 class Solution {
public:
    string XXX(vector<string>& strs) {
        //需要取strs的第i个，strs[i]，再取strs[i]中的第[j]份，因此要两次
        if(size(strs)==0){
            return "";
        }
        string res=strs[0];
        for(int i=1;i<size(strs);i++){
            int j=0;
            string &jud=strs[i];
            int l=min(res.length(),jud.length());
            while(j<l&&res[j]==jud[j]){
                j++;
            }  
            res=res.substr(0,j);
        }
         return res;
    }
};

File: Dataset/Leetcode/train/104/93.java
class Solution {
    public int XXX(TreeNode root) {
        int res = 0;
        if (root==null){
            return res;
        }

        Deque deque = new ArrayDeque();
        deque.addFirst(root);

        while (!deque.isEmpty()){
            int size = deque.size();
            for (int i=0;i<size;i++){
                TreeNode node = (TreeNode) deque.getFirst();
                deque.removeFirst();

                if (node.left!=null) deque.addLast(node.left);
                if (node.right!=null) deque.addLast(node.right);
            }
            res =res+1;

        }

        return res;
    }
}

